tests/validators/test_url.py

```diff
@@ -10,28 +12,80 @@
 
 def test_url_ok(py_and_json: PyAndJson):
     v = py_and_json(core_schema.url_schema())
-    url: Url = v.validate_test('https://example.com/foo/bar?baz=qux#quux')
-    assert isinstance(url, Url)
-    assert str(url) == 'https://example.com/foo/bar?baz=qux#quux'
-    assert repr(url) == "Url('https://example.com/foo/bar?baz=qux#quux')"
-    assert url.unicode_string() == 'https://example.com/foo/bar?baz=qux#quux'
-    assert url.scheme == 'https'
-    assert url.host == 'example.com'
-    assert url.unicode_host() == 'example.com'
-    assert url.path == '/foo/bar'
-    assert url.query == 'baz=qux'
-    assert url.query_params() == [('baz', 'qux')]
-    assert url.fragment == 'quux'
-    assert url.username is None
-    assert url.password is None
-    assert url.port == 443
+    urls: List[Url] = [
+        v.validate_test('https://example.com/foo/bar?baz=qux#quux'),
+        Url('https://example.com/foo/bar?baz=qux#quux'),
+    ]
+    for url in urls:
+        assert isinstance(url, Url)
+        assert str(url) == 'https://example.com/foo/bar?baz=qux#quux'
+        assert repr(url) == "Url('https://example.com/foo/bar?baz=qux#quux')"
+        assert url.unicode_string() == 'https://example.com/foo/bar?baz=qux#quux'
+        assert url.scheme == 'https'
+        assert url.host == 'example.com'
+        assert url.unicode_host() == 'example.com'
+        assert url.path == '/foo/bar'
+        assert url.query == 'baz=qux'
+        assert url.query_params() == [('baz', 'qux')]
+        assert url.fragment == 'quux'
+        assert url.username is None
+        assert url.password is None
+        assert url.port == 443
 
 
 @pytest.fixture(scope='module', name='url_validator')
 def url_validator_fixture():
     return SchemaValidator(core_schema.url_schema())
 
 
+class UrlValidatorMode(Enum):
+    SCHEMA_VALIDATOR = auto()
+    URL_CLASS = auto()
+    MULTI_URL_CLASS = auto()
+
+
+def validate_url(url: str, validator_mode: UrlValidatorMode, url_validator: Optional[SchemaValidator]):
```

Comment:
I don't think we need a separate method for this, or `url_test_case_helper` better to have a bit of duplication in a single function to make the tests easier to debug.

Reply:
Ok I folded this into `url_test_case_helper`